src/main/java/frc/robot/subsystems/shooter_intake/ShooterIntakeSubsystem.java

```diff
@@ -0,0 +1,78 @@
+package frc.robot.subsystems.shooter_intake;
+
+public class ShooterIntakeSubsystem {
+
+    protected State currentState = State.IDLE;
+    protected State targetState = State.IDLE;
+    ShooterIntakeIO io;
+
+    public ShooterIntakeSubsystem(ShooterIntakeIO io) {
+        this.io = io;
+        State.subsystem = this;
+    }
+
+    public void periodic() {
+
+        currentState.periodic();
+    }
+
+    public void setTargetState(State target) {
+        this.targetState = target;
+    }
+
+    // spotless:off
+    public enum State {
```

Comment:
After looking at this more, I'm not sure using the enum functions really is buying us much. It's nice that we get some inheritance here, but it's also sort of confusing to read with `changeStates` and `changeState` having two different outcomes.

One thing that I think is unfortunate is that we have to have a static instance of the subsystem inside the enum. This is not intuitive and I don't think we want to design things this way.

I think we should go back to the system we did last year for now - an enum that defines state and a switch statement that dispatches specific functions which each run the logic for their respective state.

Reply:
I believe I can fix both of those issue very quickly if there is not other issues is it fine to keep this implementation with a bit of cleaning up.